### ISSUE TITLE
[Affiliate] Referral address is removed from the local storage after successful Wrap/Approve transaction

### ISSUE THREAD

**Issue by elena-zh** [gnosis/cowswap#1869]
1. Share a referral code
2. Open the referral code by a user
3. Run successful Wrap/Approve transaction

**AR**: referral code is removed from the local stirage
https://watch.screencastify.com/v/yz12UmrqgX2y2MIodjOQ

**ER**: referral code should not be removed after Wrap/Approve transaction
(only after a successful order)

**Comment by elena-zh** [comment]
1. I shared a referral link https://cowswap.staging.gnosisdev.com/#/?referral=0xFF714b8b0e2700303eC912BD40496C3997ceEa2b to the `0x7a4d4872Bdad98DAC4BF4276186FBD0959DAA07F` account. 

2. I got the 'valid affiliate code' message.
3. Then, I did a wrap transaction --> referral address was removed from my local storage
4. I did several Expired transaction, then I was able to get a successful sell transaction: https://protocol-explorer.staging.gnosisdev.com/address/0x7a4d4872bdad98dac4bf4276186fbd0959daa07f
5. Total trades section was updated accordingly.
6. But I do not see referral info for the account `0xFF714b8b0e2700303eC912BD40496C3997ceEa2b`
![image](https://user-images.githubusercontent.com/70885163/142201542-bc4b52e3-128a-4871-b2f9-6a1378cad5f8.png)

**Comment by josojo** [comment]
`0x7a4d4872Bdad98DAC4BF4276186FBD0959DAA07F` was first referred by another user:

![image](https://user-images.githubusercontent.com/7348154/142204460-c9ae09d0-19ea-470d-b692-71b7e1dc6764.png)

**Comment by elena-zh** [comment]
More than strange to see this message then 
![image](https://user-images.githubusercontent.com/70885163/142205645-2aca9b3b-d5d2-457b-8286-32b4c60dd472.png)


However, I might assume that this referral address was shared to the account when testing in a Dev server, as the account has a successful trade there
https://protocol-explorer.dev.gnosisdev.com/address/0x7a4d4872Bdad98DAC4BF4276186FBD0959DAA07F

Anyways, if a referral address from the Dev is valid for Prod, we should not show 'valid referral address' message

**Comment by elena-zh** [comment]
> `0x7a4d4872Bdad98DAC4BF4276186FBD0959DAA07F` was first referred by another user:
> 
> ![image](https://user-images.githubusercontent.com/7348154/142204460-c9ae09d0-19ea-470d-b692-71b7e1dc6764.png)

And yes, I see on UI the referral data for this account... 
![image](https://user-images.githubusercontent.com/70885163/142207434-77d810f4-73eb-437c-a6f8-8e0de3d8c9b8.png)

**Comment by elena-zh** [comment]
I reported a separate issue for the case when referral data from prod server was added to referral address in dev server #1871 

The current issue will be related to data removal  from a local storage after transactions.

**Comment by alfetopito** [comment]
Yes, this sounds like a new bug, but it's unrelated to the changes made.
I expect you should be able to reproduce this on production.

Given that the fix is only dealing with visibility, I'd say we can release without it and address is in another release/hotfix

**Comment by elena-zh** [comment]
The issue is still reproducible.
